pageserver/src/span.rs

```diff
@@ -0,0 +1,49 @@
+#[cfg(debug_assertions)]
+use utils::tracing_span_assert::{check_fields_present, MultiNameExtractor};
+
+#[cfg(debug_assertions)]
+static TENANT_ID_EXTRACTOR: once_cell::sync::Lazy<MultiNameExtractor<1>> =
+    once_cell::sync::Lazy::new(|| MultiNameExtractor::new("TenantId", ["tenant_id"]));
+#[cfg(debug_assertions)]
+static SHARD_ID_EXTRACTOR: once_cell::sync::Lazy<MultiNameExtractor<1>> =
+    once_cell::sync::Lazy::new(|| MultiNameExtractor::new("ShardId", ["shard_id"]));
+#[cfg(debug_assertions)]
+static TIMELINE_ID_EXTRACTOR: once_cell::sync::Lazy<MultiNameExtractor<1>> =
+    once_cell::sync::Lazy::new(|| MultiNameExtractor::new("TimelineId", ["timeline_id"]));
+
+#[cfg(not(debug_assertions))]
+pub(crate) fn debug_assert_current_span_has_tenant_id() {}
+
+#[cfg(debug_assertions)]
+#[track_caller]
+pub(crate) fn debug_assert_current_span_has_tenant_id() {
+    if let Err(missing) = check_fields_present!([&*TENANT_ID_EXTRACTOR, &*SHARD_ID_EXTRACTOR]) {
+        panic!("missing extractors: {missing:?}")
+    }
+}
```

Comment:
```suggestion
#[track_caller]
pub(crate) fn debug_assert_current_span_has_tenant_id() {
    #[cfg(debug_assertions)]
    if let Err(missing) = check_fields_present!([&*TENANT_ID_EXTRACTOR, &*SHARD_ID_EXTRACTOR]) {
        panic!("missing extractors: {missing:?}")
    }
}
```

Is there a benefit in doing the two versions one?

Reply:
Didn't think about it. But, would the `track_caller` make a difference?

Reply:
No, it's effect is compile time only.

Reply:
The benefit of this suggestion doesn't warrant the time I spend clicking retry on the CI, sadly -,-

Reply:
Will shove this into https://github.com/neondatabase/neon/issues/6638